e2e-test-app/src/main/java/com/sap/ai/sdk/app/controllers/OpenAiController.java

```diff
@@ -38,6 +48,101 @@ public static OpenAiChatCompletionOutput chatCompletion() {
     return OpenAiClient.forModel(GPT_35_TURBO).chatCompletion(request);
   }
 
+  /**
+   * Asynchronous stream of an OpenAI chat request
+   *
+   * @return the emitter that streams the assistant message response
+   */
+  @SuppressWarnings("unused") // The end-to-end test doesn't use this method
+  @GetMapping("/streamChatCompletionDeltas")
+  @Nonnull
+  public static ResponseEntity<ResponseBodyEmitter> streamChatCompletionDeltas() {
+    final var request =
+        new OpenAiChatCompletionParameters()
+            .setMessages(
+                List.of(
+                    new OpenAiChatUserMessage()
+                        .addText(
+                            "Can you give me the first 100 numbers of the Fibonacci sequence?")));
+
+    final var stream = OpenAiClient.forModel(GPT_35_TURBO).streamChatCompletionDeltas(request);
+
+    final var emitter = new ResponseBodyEmitter();
+
+    final Runnable consumeStream =
+        () -> {
+          final var totalOutput = new OpenAiChatCompletionOutput();
+          // try-with-resources ensures the stream is closed
+          try (stream) {
+            stream
+                .peek(totalOutput::addDelta)
+                .forEach(delta -> send(emitter, delta.getDeltaContent()));
+          } finally {
+            send(emitter, "\n\n-----Total Output-----\n\n" + objectToJson(totalOutput));
```

Comment:
(Comment/Minor)

It could be worthwhile to not just dump the object-to-string. But instead to extract the relevant information and write it piece-by-piece. That would also demonstrate the actual benefit of the feature. Right now the interested reader would wunder.. _"what does total output even mean?"_